py/dml/dmlc.py

```diff
@@ -502,11 +508,13 @@ def show_illegal_attributes(option, opt, value, parser):
             else:
                 f = open(options.makedep, 'w')
             with f:
-                # rebuild from DML if imported file has changed
-                f.write('%s %s.c : %s\n' % (
-                    options.makedep, outputbase, deps))
-                # if imported file vanished, survive and rebuild
-                f.write('%s :\n' % (deps,))
+                    # rebuild from DML if imported file has changed
+                if options.depninja:
+                    f.write('%s.c : %s\n' % (outputbase, deps))
+                else:
+                    f.write('%s %s.c : %s\n' % (options.makedep, outputbase, deps))
+                    # if imported file vanished, survive and rebuild
+                    f.write('%s :\n' % (deps,))
```

Comment:
I did manual tests for both Ninja and Unix Makefiles generators in CMake. When using Makefiles, `--dep` works ok. Ninja generators complain about the extra target for .dmldep file and extra targets for individual files.

Reply:
I would prefer to make the args less tied to particular build systems, instead of more. The current --dep option is tailored for one particular build system (`module.mk` interpreted by GNU make). If we need something more customized, then perhaps we should extract separate options similar to gcc's `-MP` and `-MT`.

For instance:
* add `--no-phony-dep`, corresponding to -MP. (perhaps we'll later add --phony-dep and invert the default)
* add `--dep-target` for -MT, with gcc-like semantics (preserving the current default of two targets).